nebula_hw_interfaces/src/nebula_hesai_hw_interfaces/hesai_hw_interface.cpp

```diff
@@ -3130,78 +1362,30 @@ HesaiStatus HesaiHwInterface::CheckAndSetConfig()
 #ifdef WITH_DEBUG_STDOUT_HESAI_HW_INTERFACE
   std::cout << "Start CheckAndSetConfig!!" << std::endl;
 #endif
-  if (true) {
-    std::thread t([this] {
-      GetConfig(  // ctx,
-        [this](HesaiConfig & result) {
-          std::stringstream ss;
-          ss << result;
-          PrintInfo(ss.str());
-          CheckAndSetConfig(
-            std::static_pointer_cast<HesaiSensorConfiguration>(sensor_configuration_), result);
-        });
-    });
-    t.join();
-
-    std::thread t2([this] {
-      GetLidarRange(  // ctx,
-        [this](HesaiLidarRangeAll & result) {
-          std::stringstream ss;
-          ss << result;
-          PrintInfo(ss.str());
-          CheckAndSetConfig(
-            std::static_pointer_cast<HesaiSensorConfiguration>(sensor_configuration_), result);
-        });
-    });
-    t2.join();
-  } else if (false) {
-    GetConfig([this](HesaiConfig & result) {
-      std::cout << result << std::endl;
-      CheckAndSetConfig(
-        std::static_pointer_cast<HesaiSensorConfiguration>(sensor_configuration_), result);
-    });
-    GetLidarRange([this](HesaiLidarRangeAll & result) {
-      std::cout << result << std::endl;
-      CheckAndSetConfig(
-        std::static_pointer_cast<HesaiSensorConfiguration>(sensor_configuration_), result);
-    });
-    std::cout << "make thread t2" << std::endl;
-    std::thread t2([this] {
-      tcp_driver_->GetIOContext()->restart();
-      tcp_driver_->run();
-    });
-    std::cout << "made thread t2" << std::endl;
-    t2.join();
-    std::cout << "joined thread t2" << std::endl;
+  std::thread t([this] {
+    auto result = GetConfig();
 
-  } else {
-    bool stopped = tcp_driver_->GetIOContext()->stopped();
-    std::cout << "stopped: " << stopped << std::endl;
-    if (stopped) tcp_driver_->GetIOContext()->restart();
-    GetConfig(
-      [this](HesaiConfig & result) {
-        std::cout << result << std::endl;
-        CheckAndSetConfig(
-          std::static_pointer_cast<HesaiSensorConfiguration>(sensor_configuration_), result);
-      },
-      false);
-    if (stopped) tcp_driver_->run();
-    stopped = tcp_driver_->GetIOContext()->stopped();
-    std::cout << "stopped2: " << stopped << std::endl;
-    if (stopped) tcp_driver_->GetIOContext()->restart();
-    GetLidarRange(
-      [this](HesaiLidarRangeAll & result) {
-        std::cout << result << std::endl;
-        CheckAndSetConfig(
-          std::static_pointer_cast<HesaiSensorConfiguration>(sensor_configuration_), result);
-      },
-      false);
-    if (stopped) tcp_driver_->run();
-  }
+    std::stringstream ss;
+    ss << result;
+    PrintInfo(ss.str());
+    CheckAndSetConfig(
+      std::static_pointer_cast<HesaiSensorConfiguration>(sensor_configuration_), result);
+  });
+  t.join();  // here
```

Comment:
(pedantic) here?

Reply:
Addressed in [d56b77e](https://github.com/tier4/nebula/pull/126/commits/d56b77e2adffe2c41f8990cfa4926608cb163fa6)